engine/runtime/src/main/java/org/enso/interpreter/service/error/ExitException.java

```diff
@@ -0,0 +1,8 @@
+package org.enso.interpreter.service.error;
+
+/** Thrown when {@code System.exit} call was called during execution. */
+public class ExitException extends RuntimeException implements ServiceException {
```

Comment:
This exception shall extend `AbstractTruffleException`, implement `InteropLibrary` and provide `isExit` and `exitCode` messages.

Reply:
In that case, we will only implement [Soft exits](https://github.com/oracle/graal/blob/master/truffle/docs/Exit.md#soft-exit). But I guess that is fine. Refactored in 9b9c8cf4a0892471cc4eba2a2824bb34675d7fcc. IDE handles the node with `System.exit 42` as a Panic and does not evaluation of other nodes:
![image](https://github.com/user-attachments/assets/516490b5-755f-453e-8dc9-744437dc51bd)